libControls/ToolTip.cpp

```diff
@@ -42,7 +43,20 @@ void ToolTip::add(Control& c, const std::string& str)
 
 void ToolTip::buildDrawParams(std::pair<Control*, std::string>& item, int mouseX)
 {
-	const auto toolTipSize = mFont.size(item.second) + PaddingSize * 2;
+	const auto toolTipLineHeight = mFont.height();
+    const auto numberOfLines = static_cast<int>(std::count(item.second.begin(), item.second.end(), '\n') + 1);
+	const auto toolTipHeight = toolTipLineHeight * numberOfLines + PaddingSize.y * 2;
+
+	std::istringstream stream(item.second);
+	std::string line;
+	int maxWidth = 0;
+	while (std::getline(stream, line, '\n'))
+	{
+		maxWidth = std::max(maxWidth, mFont.size(line).x);
+	}
+	const auto toolTipWidth = maxWidth + PaddingSize.x * 2;

```

Comment:
I like how there's a final calculated result which is marked as `const`. Though I wonder if there's a way to calculate it that doesn't "leak" mutable local variables for the temporary values throughout the remainder of the function scope.

My first thought was if there was a ranged-for syntax to process a string by lines. Didn't see anything promising.

My second thought was perhaps a regular `for` loop could be used with multiple variable initialization. That would allow some of the temporary variables to be contained within the `for` block scope. Though it seems multiple variable initialization is only possible when they are all the same type.

My final thought, is perhaps the code could be extracted to a separate function. That would keep the drawing code at a high level, while separating the low level details of finding the max width of a sequence of lines.